packages/core/upload/admin/src/components/TableList/TableRows.js

```diff
@@ -53,23 +72,39 @@ export const TableRows = ({ onEditAsset, onEditFolder, onSelectOne, rows, select
                 </Td>
               );
             })}
-            {((elementType === 'asset' && onEditAsset) ||
-              (elementType === 'folder' && onEditFolder)) && (
-              <Td>
-                <IconButton
-                  label={formatMessage({
-                    id: getTrad('control-card.edit'),
-                    defaultMessage: 'Edit',
-                  })}
-                  onClick={() =>
-                    elementType === 'asset' ? onEditAsset(element) : onEditFolder(element)
-                  }
-                  noBorder
-                >
-                  <Pencil />
-                </IconButton>
-              </Td>
-            )}
+
+            <Td {...stopPropagation}>
+              <Flex justifyContent="flex-end">
+                {elementType === 'folder' && (
+                  <IconButton
+                    forwardedAs={Link}
+                    label={formatMessage({
+                      id: getTrad('list.folders.link-label'),
+                      defaultMessage: 'Access folder',
+                    })}
+                    to={getFolderURL(pathname, query, element)}
+                    noBorder
+                  >
+                    <Eye />
+                  </IconButton>
+                )}
+                {((elementType === 'asset' && onEditAsset) ||
```

Comment:
do you actually need to check the function exists? feels overkill here.

Reply:
we could just provide a default noop func – `() => {}`

Reply:
I didn't take the time to check all permissions yet, I'll do that for my final PR
I think I could make the function required for now since I don't know any case where it shouldn't be rn 🤔

Reply:
That would work for me, makes the condition easier to read 🤓